package/MDAnalysis/analysis/density.py

```diff
@@ -443,6 +443,13 @@ def _prepare(self):
                           " defined grid will need to be "
                           "provided instead.")
                 raise ValueError(errmsg) from err
+            if any([self._xdim, self._ydim, self._zdim]):
```

Comment:
I'm going to let the other devs weigh in here too, but I'm personally not super fond of adding extra warnings in what is already a warning/error heavy piece of code.

Personally I'd favour we instead make line 403 a call to any for the dims and gridcenter, then if any of them are not set they just trigger the errors from the checks in `_set_user_grid`.

Part of my rationale here is also that if folks are setting any of these values manually and missing any of the others, then it's probably unintentional and we should be failing rather than warning.

Reply:
I see what you mean, this warning could be misleading in some cases. I will wait for response from the other devs, but I'm happy to change it.

Reply:
I agree with @IAlibay, in that the checks belong with the other grid checks in `_set_user_grid`.

Reply:
Ok let's do that then.

Reply:
@IAlibay One question, when making line 403 a call to any for the dims and gridcenter, the error "ValueError: gridcenter must be a 3D coordinate" is thrown. Would adding a specific error to `_set_user_grid` be over complicating things?
I personally think this would make things more clear.

Reply:
What'd you mean by "specific error"? I probably would just expand on that error message, just say f"gricenter is set to {gridcenter}: to use a user-defined grid, it must be a 3D coordinate" (or something better written but along those lines)?

Reply:
Oh, I was a bit unclear. I referred to the case where xyzdims are provided but the gridcenter isn't. When making line 403 a call to any for the dims and gridcenter, the above ValueError is thrown in this case. I thought that an error of the form "Grid dimensions are provided but no grid center is." would be more informative for the user. What are your thoughts on this?

Reply:
We can add an extra user input validation in `_set_user_grid` if it makes things easier to read/more intuitive. If it can be worked within the context of our existing check then that's even better (i.e. if it's a check after a `try` then the cost is near 0).